pulumitest/tempdir.go

```diff
@@ -0,0 +1,95 @@
+package pulumitest
+
+import (
+	"fmt"
+	"os"
+	"strings"
+	"sync"
+	"unicode"
+	"unicode/utf8"
+)
+
+func tempDirWithoutCleanupOnFailedTest(t PT, desc string) string {
+	t.TempDir()
+	c := getOrCreateTempDirState(t)
```

Comment:
Instead of the complexity of global state and incrementing to get a unique file name, can we use [os.MkdirTemp](https://pkg.go.dev/os#MkdirTemp)?

Reply:
This is inlined with minimal viable changes from the Go stdlib. There must be reasons why Go stdlib evolved to be the way it is.